scripts/mirror.js

```diff
@@ -183,206 +183,358 @@ const updateNotes = (notes, regex, replace) => {
+      originalData.notes,
+    );
+
+    if (newNotes) {
+      newData.notes = newNotes;
+    }
   }
 
-  let newNotes = combineNotes(
-    updateNotes(chromeData.notes, /Chrome/g, 'Edge'),
-    edgeData.notes,
-  );
+  return newData;
+};
+
+/**
+ * @param {SupportStatement} originalData
+ * @param {SupportStatement} sourceData
+ * @param {string} source
+ * @returns {SupportStatement}
+ */
+const bumpFirefoxAndroid = (originalData, sourceData, source) => {
+  let newData = sourceData;
+
+  if (typeof sourceData.version_added === 'string') {
+    newData.version_added = Math.max(
+      4,
+      Number(sourceData.version_added),
+    ).toString();
+  }
 
-  if (newNotes) {
-    newData.notes = newNotes;
+  if (
+    sourceData.version_removed &&
+    typeof sourceData.version_removed === 'string'
+  ) {
+    newData.version_removed = Math.max(
+      4,
+      Number(sourceData.version_removed),
+    ).toString();
   }
 
   return newData;
 };
 
 /**
- * @param {SupportStatement} data
- * @param {string} destination
+ * @param {SupportStatement} originalData
+ * @param {SupportStatement} sourceData
  * @param {string} source
+ * @returns {SupportStatement}
+ */
+const bumpOpera = (originalData, sourceData, source) => {
+  let newData = sourceData;
+
+  if (typeof sourceData.version_added === 'string') {
+    newData.version_added = getMatchingBrowserVersion(
+      'opera',
+      browsers[source].releases[sourceData.version_added],
+    );
+  }
+
+  if (
+    sourceData.version_removed &&
+    typeof sourceData.version_removed === 'string'
+  ) {
+    newData.version_removed = getMatchingBrowserVersion(
+      'opera',
+      browsers[source].releases[sourceData.version_removed],
+    );
+  }
+
+  if (typeof sourceData.notes === 'string') {
+    newData.notes = updateNotes(sourceData.notes, /Chrome/g, 'Opera');
+  }
+
+  return newData;
+};
+
+/**
  * @param {SupportStatement} originalData
+ * @param {SupportStatement} sourceData
+ * @param {string} source
+ * @returns {SupportStatement}
  */
-const bumpVersion = (data, destination, source, originalData) => {
-  let newData = null;
-  if (data == null) {
-    return null;
-  } else if (
-    Array.isArray(data) &&
-    !(destination == 'edge' && source == 'chrome')
+const bumpOperaAndroid = (originalData, sourceData, source) => {
+  let newData = sourceData;
+
+  if (typeof sourceData.version_added === 'string') {
+    newData.version_added = getMatchingBrowserVersion(
+      'opera_android',
+      browsers[source].releases[sourceData.version_added],
+    );
+  }
+
+  if (
+    sourceData.version_removed &&
+    typeof sourceData.version_removed === 'string'
   ) {
-    newData = [];
-    for (let i = 0; i < data.length; i++) {
-      newData[i] = bumpVersion(data[i], destination, source, originalData);
-    }
-  } else {
-    newData = {};
-    for (let i in data) {
-      newData[i] = data[i]; // Prevent shallow copy / modification of source data
```

Comment:
I think this is missing in the new bumpVersion and so source data is now actually modified.

Reply:
Urgh, this is one of my pet peeves with JavaScript and I keep forgetting...